docs/migrate-4.md

```diff
@@ -0,0 +1,203 @@
+# Migration to v4
+
+## Update `nw-builder`
+
+With npm:
+
+```shell
+npm update nw-builder@^4.0.0
+```
+
+With yarn:
+
+```shell
+yarn upgrade nw-builder@^4.0.0
+```
+
+With pnpm:
+
+```shell
+pnpm update nw-builder@^4.0.0
+```
+
+> Note: `nw-builder` has been tested on Node 16 and 18 only.
+
+## Update options
+
+Let's take an example of v3 code and migrate it to v4.
+
+```javascript
+const NwBuilder = require("nw-builder");
+
+const nw = new NwBuilder({
+  files: ["./nwapp", "./other/**/*.js"],
+  version: "latest",
+  flavor: "normal",
+  platforms: ["win32", "win64", "osx32", "osx64", "linux32", "linux64"],
+  cacheDir: "./cache",
+  buildDir: "./build",
+  buildType: "versioned",
+  forceDownload: true,
+  appName: "nwdemo",
+  appVersion: "0.1.0",
+  argv: "--nw-stderr-logging",
+  macCredits: "./nwapp/credits.html",
+  macIcns: "./nwapp/mac.icns",
+  macPlist: { ... },
+  winVersionString: { ... },
+  winIco: "./nwapp/win.ico",
+  zip: true,
+  macZip: false,
+  mergeZip: false,
+});
+
+nw.build();
+```
+
+Update the import path
+
+```patch
+-const NwBuilder = require("nw-builder");
++const { nwbuild } = require("nw-builder");
+```
+
+Replace the `NwBuilder` initialization with a function
+
+```patch
+-const nw = new NwBuilder({
++nwbuild({
+```
+
+The `files` property has been renamed to `srcDir`. As the name suggests, it does not taken in any globbing patterns. Structure of the NW app if left up to the user.
+
+```patch
+-  files: ["./nwapp", "./other/**/*.js"],
++  srcDir: "./nwapp",
+```
+
+Add the `mode` option.
+
+```patch
++  mode: "build",
+```
+
+The `platforms` option has been removed and replaced with `platform` and `arch`. Notice that one `nwbuild` function call now creates one build only. Refer to the [documentation](./index.md) for valid `platform` and `arch` values.
```

Comment:
As mentioned in issue #743, this documentation should be moved to the main `README.md` file.

Reply:
✅